python/pyspark/sql/connect/types.py

```diff
@@ -184,3 +186,113 @@ def proto_schema_to_pyspark_data_type(schema: pb2.DataType) -> DataType:
         )
     else:
         raise Exception(f"Unsupported data type {schema}")
+
+
+def _need_converter(dataType: DataType) -> bool:
```

Comment:
refer to https://github.com/apache/spark/blob/51b04406028e14fbe1986f6a3ffc67853bd82935/python/pyspark/sql/types.py#L1692-L1752 the converter in PySpark

Reply:
Can we call that method directly? e.g.)

```
if isinstance(dataType, MapType):
    # Different from PySpark, here always needs conversion,
    # since the input from Arrow is a list of tuples.
    return True
else:
    return _need_converter(dataType)
```